siliconcompiler/remote/client.py

```diff
@@ -136,24 +136,25 @@ def _remote_preprocess(chip, steplist):
         # Remove each local step from the list of steps to run on the server side.
         remote_steplist.remove(local_step)
 
-        # Need to override steplist here to make sure check_manifest() doesn't
-        # check steps that haven't been setup.
-        chip.set('option', 'steplist', local_step)
-
-        # Run the actual import step locally with multiprocess as _runtask must
-        # be run in a separate thread.
-        # We can pass in an empty 'status' dictionary, since _runtask() will
-        # only look up a step's dependencies in this dictionary, and the first
-        # step should have none.
-        run_task = multiprocessor.Process(target=chip._runtask,
-                                          args=(local_step, index, {}))
-        run_task.start()
-        run_task.join()
-        if run_task.exitcode != 0:
-            # A 'None' or nonzero value indicates that the Process target failed.
-            ftask = f'{local_step}{index}'
-            chip.error(f"Could not start remote job: local setup task {ftask} failed.",
-                       fatal=True)
+        if not chip.get('option', 'resume'):
```

Comment:
what happens if resume is on, but the remoteid is not set?

Reply:
It will upload the current local run to be resumed on the remote server.